src/AngleSharp.Diffing.Tests/Core/HtmlDifferenceEngineTest.cs

```diff
@@ -357,6 +357,40 @@ public void Test2()
             results.ShouldBeEmpty();
         }
 
+        [Fact(DisplayName = "When comparer returns SkipChildren from an element comparison, child nodes are not compared")]
+        public void Test3()
+        {
+            var sut = CreateHtmlDiffer(
+                nodeMatcher: OneToOneNodeListMatcher,
+                nodeFilter: NoneNodeFilter,
+                nodeComparer: c => c.Control.Node.NodeName == "P" ? CompareResult.Same | CompareResult.SkipChildren : throw new Exception("NODE COMPARER SHOULD NOT BE CALLED ON CHILD NODES"),
+                attrMatcher: AttributeNameMatcher,
+                attrFilter: NoneAttrFilter,
+                attrComparer: SameResultAttrComparer
+                );
+
+            var results = sut.Compare(ToNodeList(@"<p><em>foo</em></p>"), ToNodeList(@"<p><span>baz</span></p>"));
+
+            results.ShouldBeEmpty();
+        }
+
+        [Fact(DisplayName = "When comparer returns SkipAttributes from an element comparison, attributes are not compared")]
+        public void Test4()
+        {
+            var sut = CreateHtmlDiffer(
+                nodeMatcher: OneToOneNodeListMatcher,
+                nodeFilter: NoneNodeFilter,
+                nodeComparer: c => CompareResult.Same | CompareResult.SkipAttributes,
+                attrMatcher: AttributeNameMatcher,
+                attrFilter: NoneAttrFilter,
+                attrComparer: SameResultAttrComparer
+                );
+
+            var results = sut.Compare(ToNodeList(@"<p id=""foo""></p>"), ToNodeList(@"<p id=""bar"" unexpected></p>"));
+
+            results.ShouldBeEmpty();
+        }
```

Comment:
Should this be a theory with the combinations of `CompareResult.SkipAttributes` and the other `CompareResult` as input to make sure the test verifies all the possible outcomes (this also applies to `Test3`)?

E.g.:

```
CompareResult.Same | CompareResult.SkipAttributes
CompareResult.Different | CompareResult.SkipAttributes
CompareResult.Skip| CompareResult.SkipAttributes
```

Reply:
I'v changed to theory. But in case `CompareResult.Different | CompareResult.SkipAttributes` compare result doesn't empty. I skipped this test.